scripts/genMaterial.js

```
@@ -0,0 +1,622 @@
+  const regex = /slots\.(\w+)/g;
+  const slotNames = new Set(); // 使用 Set 去重
+
+  let match;
+  while ((match = regex.exec(funcString)) !== null) {
+    slotNames.add(match[1]); // 添加 x 到集合中
+  }
+
+  return Array.from(slotNames); // 转换为数组
+}
+
+function getLabelName(key, labelName) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)